### PR TITLE
Update npm dependencies

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

Mostly just updating npm dependencies. One breaking change that affected unit tests slightly.

The big change is deduping the node_modules of angular-pouch so that it uses the pouchdb-browser version defined in the root package.json (6.3.4) rather than it's own version (6.0.5). This means we're actually getting the pouchdb we thought we were.

medic/medic-webapp#3776

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

File: tests/karma/unit/services/send-message.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ describe('SendMessage service', function() {
     chai.expect(msg.from).to.equal(expected.from);
     chai.expect(msg.sent_by).to.equal(expected.sent_by);
     chai.expect(msg.to).to.equal(expected.to);
-    chai.expect(msg.uuid).to.be.defined; // jshint ignore:line
+    chai.expect(msg.uuid).to.not.be.undefined; // jshint ignore:line
```
Comment (alxndrsn) [inline tests/karma/unit/services/send-message.js]: :~(

**Comment by alxndrsn** [comment]
For some reason I thought all of our web-based dependencies were already "de-duped" when the app gets bundled.  If not, does that mean we have multiple versions of multiple different libraries being deployed to clients?

I've restarted the failing travis build; if that passes then this looks fine to merge.

**Comment by garethbowen** [comment]
@alxndrsn Dependencies are deduped if there is a version that satisfies both ranges. For example, if we depend on pouchdb v1.1+ and angular-pouch depends on pouchdb v1.3+ then npm will only download the v1.3 (or higher) which satisfies both ranges. In this case we depend on `^6.3.4` (any version that starts with a 6 and is greater than 6.3.4) and angular-pouch depends on `6.0.4` (exact match required). As there is no way for npm to satisfy both ranges it downloads two copies. This is by design (and the right thing to do) because npm has no way to know if angular-pouch can work with a later version of pouchdb.

In my opinion the angular-pouch module should definitely not be tied to an exact pouchdb version. At the very least it should pull the latest patch, but probably any backwards compatible minor version would be fine, so specifying `^6.0.4` would be fine and would mean we could dictate the version we want to use.

To further complicate matters this is just talking about npm. When we browserify we may or may not get both versions in the bundle. If we always require angular-pouch then npm will always return the same version and we'll only get one copy of pouch in the final bundle. If we sometimes require pouchdb directly then we will get both versions.

It'd be really interesting to investigate how many duplicates we have in the bundle...

**Comment by alxndrsn** [comment]
> In my opinion the angular-pouch module should definitely not be tied to an exact pouchdb version.

Fully agree.  Have you submitted a PR upstream?  In the meantime I'd probably prefer patching their package.json than deleting the bad pouchdb from `node_modules`, but it's not that important.

> It'd be really interesting to investigate how many duplicates we have in the bundle...

And this.  The following may give some helpful info:
```
rm -rf node_modules && npm install --production && npm dedupe && \
  find . -name node_modules | xargs -n1 ls -1 | sort | uniq -d
```
Gives me:
```
assert-plus
async
debug
lie
ms
pouchdb-browser
spark-md5
```

One copy of `async` is required by `uglify-js`, which is required by `nools`…

Obviously this doesn't quite tell us what browserify ends up bundling.  I can't find built-in support in browserify for generating a report on this, although it seems an obvious feature.

**Pull request merged**